### PR TITLE
feat:  materialize executor support ignore conflict behavior

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
This PR allow materialize executor ignore incoming pk conflict, which change `handle_pk_conflict` to `conflict_behavior`. Ignoring conflict is useful for append only source.
close https://github.com/risingwavelabs/risingwave/issues/8116

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by st1page — COMMENTED** [review]
maybe we can extract the function `fn handle_conflict(key, row_op, old_row)` to reuse the code.

File: Cargo.lock [hunk 1]
```diff
@@ -2514,9 +2514,9 @@ checksum = "33ab36ccf1ed3d9c719f4c8725b34482c530f6d569c084ea762490b5e19f77bf"
 
 [[package]]
 name = "generator"
-version = "0.7.2"
+version = "0.7.3"
```
Comment (xxchan) [inline Cargo.lock:2517]: Why cargo.lock is changed?

File: src/stream/src/executor/mview/materialize.rs [hunk 2]
```diff
@@ -196,7 +192,37 @@ impl<S: StateStore> MaterializeExecutor<S> {
                                 None => continue,
                             }
                         }
-                        false => {
+                        ConflictBehavior::IgnoreConflict => {
```
Comment (st1page) [inline src/stream/src/executor/mview/materialize.rs:195]: please reuse the code in `ConflictBehavior::OverWrite` and `ConflictBehavior::IgnoreConflict` branch and the only difference here is just one line code.

**Review by st1page — COMMENTED** [review]
`ignore_conflict` and `overwrite_conflict` can merge into one function which accept a ConflictBehavior :thinking

File: src/stream/src/executor/mview/materialize.rs (hunk 2 above)
Comment (wcy-fdu) [reply src/stream/src/executor/mview/materialize.rs:195]: L147 `let mut input = self.input.execute();` takes the ownership of `self`, so it's difficult to reuse the `fill_buffer_from_chunk`  and `self.state_table.write_chunk()` in one external function.

**Review by st1page — APPROVED** [review]
LGTM

File: proto/stream_plan.proto [hunk 3]
```diff
@@ -181,8 +181,14 @@ message MaterializeNode {
   repeated plan_common.ColumnOrder column_orders = 2;
   // Used for internal table states.
   catalog.Table table = 3;
-  // Used to control whether doing sanity check, open it when upstream executor is source executor.
-  bool handle_pk_conflict = 4;
+  // Used to handle pk conflict, open it when upstream executor is source executor.
+  HandleConflictBehavior handle_pk_conflict_behavior = 4;
+}
+
+enum HandleConflictBehavior {
```
Comment (BugenZhao) [inline proto/stream_plan.proto:188]: Why do we get two definitions of the same `enum`? 👀

**Pull request merged**